nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/renditions/RenditionsObject.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Thomas Roger
+ *     Nour Al Kotob
+ */
+
+package org.nuxeo.rest.management.renditions;
+
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.core.bulk.BulkService;
+import org.nuxeo.ecm.core.bulk.message.BulkCommand;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus;
+import org.nuxeo.ecm.core.bulk.message.BulkStatus.State;
+import org.nuxeo.ecm.platform.picture.recompute.RecomputeViewsAction;
+import org.nuxeo.ecm.platform.thumbnail.action.RecomputeThumbnailsAction;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.kv.KeyValueService;
+import org.nuxeo.runtime.kv.KeyValueStore;
+
+/**
+ * @since 11.1
+ */
+@WebObject(type = "renditions")
+public class RenditionsObject extends DefaultObject {
+
+    protected Set<State> notAtWork = EnumSet.of(State.UNKNOWN, State.ABORTED, State.COMPLETED);
+
+    protected Lock picturesLock = new ReentrantLock();
+
+    protected Lock thumbnailsLock = new ReentrantLock();
+
+    public static final String RECOMPUTE_VIEWS_ACTION_COMMAND_ID = RecomputeViewsAction.ACTION_NAME + "_commandId";
+
+    public static final String RECOMPUTE_THUMBNAILS_ACTION_COMMAND_ID = RecomputeThumbnailsAction.ACTION_NAME
+            + "_commandId";
+
+    public static final String THUMBNAILS_DEFAULT_QUERY = "SELECT * FROM Document WHERE ecm:mixinType = 'Thumbnail' AND thumb:thumbnail/data IS NULL AND ecm:isVersion = 0 AND ecm:isProxy = 0 AND ecm:isTrashed = 0";
+
+    public static final String PICTURES_DEFAULT_QUERY = "SELECT * FROM Document WHERE ecm:mixinType = 'Picture' AND picture:views/*/title IS NULL";
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("pictures/recompute")
+    public BulkStatus doPostPictures() throws OperationException {
+        BulkService bulkService = Framework.getService(BulkService.class);
+        String commandId = getCommandId(RECOMPUTE_VIEWS_ACTION_COMMAND_ID);
+        if (picturesLock.tryLock()) {
```

Comment:
This lock is not cluster-wide. Don't know if it's something we relaxed but it should be considered/discussed (I may have miss something).

To have a cluster-wide lock, we can improve (or get inspired by) `LockHelper`.

Reply:
Yes there was a missunderstanding. I thought we went for the `ReentrantLock`
I implemented it with LockHelper and Suppliers so I can get the resulting `BulkStatus`

Reply:
Great, worth to check with @efge if we can directly use `LockHelper` as it's using `document-lock` KVStore. We may want to not put management locks in the same KV than document.